docs/dev-sdk.md

```diff
@@ -0,0 +1,60 @@
+# SDK Development
+This document will highlight how to add new APIs to the OpenStorage SDK.
+
+## SDK Requirements
+All SDK APIs and values must satisfy by the following:
+
+* Must be readable
+    * SDK APIs and values must be concrete clear values
+    * `string` values are for ids or strings. They are not meant to marshal information as a `yaml`, create a concrete _message_ instead.
+* No value options passed as `string`
+    * Instead of passing "Done", or "paused", use enums for these value. Making it clear to the reader.
+* Services:
+    * Services should be in the format `OpenStorage<Type>`.
+    * Note that the service is a collection of APIs and are grouped as such in the documentation.
+        * Here is an example for [OpenStorageClusterService](https://libopenstorage.github.io/w/generated-api.html#openstorageapiopenstoragecluster)
+* APIs
+    * If it is a new service, then it should have `Create`, `Inspect`, `Delete`, or `Enumerate` style APIs.
+    * All APIs **must** have a single message for the request and a single message for the response with the following style: `Sdk<Service Type><Api Name>Request|Response`
+* Enums
+    * All enums must be unique across the entire proto file, not just the single enum.
+* Messages
+    * Try not to use `uint64`. Instead try to use signed `int64`. (There is a reason for this which is why CSI changed all uint64s to in64s in version 0.2, but I can find out why.)
```

Comment:
is this good point to start discussing about backword compatibility or versioning for SDK?. Useful resources https://groups.google.com/forum/#!topic/grpc-io/LPsPg5ctQd4

Reply:
do we need to specify current REST endpoint issue and similar current issues,  if we are not addressing them in initial release?
https://github.com/libopenstorage/openstorage/issues/452